### PR TITLE
Mistralai mistral 7 b v0 1

### PR DESCRIPTION
**Pull request by Suvratpalt**
## Finetune

FT on text-gen qna: https://ml.azure.com/experiments/id/2ca09003-5551-4c26-ac57-85f9cf5bb709/runs/olden_kiwi_68nf2gv2qw?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-australiaeast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

### FT model deployment results

ep: https://ml.azure.com/endpoints/realtime/mistral-ai-ft-model-deploy?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-australiaeast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

## Evaluation

evaluation run link: https://ml.azure.com/runs/eb500a2e-33fa-4da7-a387-c1d023d04e46?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/v-kartik2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

## Inference

Batch endpoint: https://ml.azure.com/endpoints/batch/mistralai-mistral-7b/detail?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/model-validation-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

Deployment Endpoint link: https://ml.azure.com/endpoints/realtime/mistral-7b-endpoint/test?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/donotuseordelete-model-validation-rg/providers/Microsoft.MachineLearningServices/workspaces/model-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

image (2)

## Model details

model save to blobstorage: https://ml.azure.com/runs/brave_deer_yt115gsjnw?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/suvrat&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

**conda.yaml:**

```
channels:
- conda-forge
dependencies:
- python=3.8.17
- pip<=23.2.1
- pip:
  - mlflow==2.6.0
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.6.0
  - azureml-core==1.52.0
  - azureml-mlflow==1.52.0
  - azureml-metrics==0.0.27
  - scikit-learn==1.1.0
  - cryptography==41.0.2
  - python-dateutil==2.8.2
  - datasets==2.14.4
  - soundfile==0.12.1
  - librosa==0.10.0.post2
  - diffusers==0.19.3
  - sentencepiece==0.1.99
  - transformers==4.34.0
  - torch==2.0.0
  - accelerate==0.21.0
  - Pillow==9.4.0
  - azureml-evaluate-mlflow==0.0.27
- pycocotools=2.0.4
name: mlflow-env
```

**MLmodel:**

```
flavors:
  hftransformersv2:
    code: null
    hf_config_class: AutoConfig
    hf_pretrained_class: MistralForCausalLM
    hf_tokenizer_class: AutoTokenizer
    huggingface_id: mistralai/Mistral-7B-v0.1
    model_data: data
    pytorch_version: 2.0.0+cu117
    task_type: text-generation
    transformers_version: 4.34.0
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.17
mlflow_version: 2.6.0
model_uuid: f22601a45680448690a2de63b1e86bc2
utc_time_created: '2023-10-02 03:24:50.844826'
```

### CODE REVIEW

File: assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  evaluation-recommended-sku: Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  languages: EN
+  azureml.copyRegistryFilesToWorkspace: true
+tags:
+  Featured: ''
+  Preview: ''
+  evaluation_compute_allow_list:
+  - Standard_NC12s_v3
```
Comment (novaturient95) [inline assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml:16]: Please add CPU ones similar to here:
https://github.com/Azure/azureml-assets/pull/1397/files

Comment (novaturient95) [reply assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml:16]: online inference was tested on : Standard_E16s_v3. Please add this as well . though inference is not that great we can keep this

File: assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: mistralai-Mistral-7B-Instruct-v0-1
+path: ./
+properties:
+  SHA: 3dc28cf29d2edd31a0a7b8f0b21637059815b4d5
```
Comment (novaturient95) [inline assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml]: this has been copied from instruct :) 
lets update with what is relevant here

File: assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: mistralai-Mistral-7B-Instruct-v0-1
```
Comment (novaturient95) [inline assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml]: here as well

File: assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml (hunk 1 above)
Comment (Suvratpalt) [reply assets/models/system/mistralai-Mistral-7B-v0-1/spec.yaml:16]: done

**Review by novaturient95 — APPROVED** [review]

**Review by RSavage2 — DISMISSED** [review]

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  Featured: ''
+  Preview: ''
+  evaluation_compute_allow_list:
+  - Standard_NC12s_v3
+  - Standard_NC24s_v3
+  - Standard_NC24rs_v3
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  inference_compute_allow_list:
```
Comment (skasturi) [inline assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:24]: Let's more add SKUs here. Specifically, please include all the  A100 SKUs. 

Standard_NC48ads_A100_v4
Standard_NC96ads_A100_v4
Standard_ND96asr_v4
Standard_ND96amsr_A100_v4
Standard_NC24ads_A100_v4

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  model_specific_defaults:
+    precision: '16'
+    deepspeed_stage: '3'
+    apply_deepspeed: 'true'
+    apply_ort: 'true'
+  license: apache-2.0
+  task: text-generation
+  SharedComputeCapacityEnabled: ''
```
[the remark this answers is not in the document]
Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:43]: Done

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml (hunk 4 above)
Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:24]: Thank you. Added.

File: assets/models/system/Mistral-7B-v0-1/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: Mistral-7B-Instruct-v0-1
```
Comment (skasturi) [inline assets/models/system/Mistral-7B-v0-1/spec.yaml:2]: Mistral-7B-Instruct-v01

Comment (novaturient95) [reply assets/models/system/Mistral-7B-v0-1/spec.yaml]: updated to `Mistral-7B-v01` for base model

**Review by skasturi — APPROVED** [review]

**Review by skasturi — APPROVED** [review]

**Pull request merged**